test/byContract/IexecPoco/03_reveal.test.ts

```diff
@@ -0,0 +1,291 @@
+            expect(task.revealCounter).equal(1);
+            expect(task.resultDigest).equal(resultDigest);
+        });
+
+        it('Should not reveal when task is not in revealing status', async () => {
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.status).equal(TaskStatusEnum.ACTIVE);
+
+            await expect(
+                iexecPocoAsWorker.reveal(taskId, resultDigest),
+            ).to.be.revertedWithoutReason(); // require#1
+        });
+
+        it('Should not reveal after deadline', async () => {
+            await iexecPocoAsWorker
+                .contribute(
+                    taskId,
+                    resultHash,
+                    resultSeal,
+                    emptyEnclaveAddress,
+                    emptyEnclaveSignature,
+                    schedulerSignature,
+                )
+                .then((tx) => tx.wait());
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.status).equal(TaskStatusEnum.REVEALING);
+            await time.setNextBlockTimestamp(task.revealDeadline);
+            // revealing task
+            // but after deadline
+            await expect(
+                iexecPocoAsWorker.reveal(taskId, resultDigest),
+            ).to.be.revertedWithoutReason(); // require#2
+        });
+
+        it('Should not reveal twice', async () => {
+            await iexecPocoAsWorker
+                .contribute(
+                    taskId,
+                    resultHash,
+                    resultSeal,
+                    emptyEnclaveAddress,
+                    emptyEnclaveSignature,
+                    schedulerSignature,
+                )
+                .then((tx) => tx.wait());
+            const task = await iexecPoco.viewTask(taskId);
+            expect(task.status).equal(TaskStatusEnum.REVEALING);
+            await iexecPocoAsWorker.reveal(taskId, resultDigest).then((tx) => tx.wait());
+            const contribution = await iexecPoco.viewContribution(taskId, worker.address);
+            expect(contribution.status).equal(ContributionStatusEnum.PROVED);
+            // revealing task, before deadline
+            // but contribution status not contributed anymore (since already proved)
+            await expect(
+                iexecPocoAsWorker.reveal(taskId, resultDigest),
+            ).to.be.revertedWithoutReason(); // require#3
+        });
+
+        it('Should not reveal when outside consensus', async () => {
+            const { dealId, taskIndex, taskId } = await iexecWrapper.signAndMatchOrders(
+                buildOrders({
+                    assets: ordersAssets,
+                    requester: requester.address,
+                    prices: ordersPrices,
+                    volume,
+                    trust: 3,
+                    tag: standardDealTag,
+                    salt: ethers.utils.hexZeroPad('0x' + Date.now().toString(), 32), // make all
```

Comment:
Do we really need to do that? Request and workerpool orders are unique here since you changed the trust value.

Reply:
Let me be more explicit yes! [Improve salt comment](https://github.com/iExecBlockchainComputing/PoCo/pull/114/commits/8bffaa54fe2c38efeee45939258125a1d3b319c5)